### ISSUE TITLE
extracting to fileshare not supported

### ISSUE THREAD

**Issue by mvanwaaijen** [quentinrossetti/node-7z#52]
Hi,

Is it possible to support the extraction of files from and to fileshares?

If I use zip.ExtractFull ('//servername/sharename/folder/zipfile.zip', '//servername/sharename/folder'), I get "The system cannot find the path specified". Extracting it to a local path works fine.

Is it possible to add the support for UNCs?

Thanks,

**Comment by mvanwaaijen** [comment]
I found the reason: the path.normalize in util/run.js will replace the double slashes with a single slash because for the input file there is a leading blank and surrounding quotes and for the output folder there is -o in front of it. 

here is the code with comments that I had modified slightly in run.js:

```javascript
    // Parse and add command (non-switches parameters) to `args`.
    // var regexpCommands = /"((?:\\.|[^"\\])*)"/g;
    var regexpCommands = /\s"((?:\\.|[^"\\])*)"/g; // added \s to only match input zipfile and ignore output file
    var commands       = command.match(regexpCommands);
    if (commands) {
      commands.forEach(function (c) {
        c = c.replace(/\//, path.sep);
        c = c.replace(/\\/, path.sep);
        c = c.trimLeft(); // added to trim leading spaces. path.normalize can't handle that.
        c = c.replace(/"/g, ''); // added to remove quotes as path.normalize can't handle that.
        c = path.normalize(c);
        args.push('"' + c + '"'); // add back the quotes to cover for files with spaces in the name
      });
    }

    // Special treatment for the output switch because it is exposed as a
    // parameter in the API and not as a option. Plus wilcards can be passed.
    var regexpOutput = /-o"((?:\\.|[^"\\])*)"/g;
    // var output       = command.match(regexpOutput); // use regex.exec instead to get a full matches collection
    var output = regexpOutput.exec(command); // use regex.exec to also get the groups instead of only the full match
    if (output) {
      // args.pop(); // the regexpCommands is ignoring the output parameter, so no need to pop it anymore
      // var o = output[0]; // replace by output[1]
      var o = output[1]; // take just the file name without the -o prefix as path.normalize will screw up a UNC path because of that
      o = o.replace(/\//, path.sep);
      o = o.replace(/\\/, path.sep);
      o = o.replace(/"/g, '');
      o = '-o' + path.normalize(o); // put back the -o prefix after normalizing the output folder path
      args.push(o);
    }
```

**Comment by quentinrossetti** [comment]
Closed in `v1.0.0`
Check the README to adapt your code to the new API
